edxval/models.py

```diff
@@ -199,12 +203,51 @@ def deconstruct(self):
         return name, path, args, kwargs
 
 
+class ListField(models.CharField):
+    """
+    ListField use to store and retrieve list data.
+    """
+    __metaclass__ = models.SubfieldBase
+
+    def get_prep_value(self, value):
+        """
+        Converts a list to its json represetation to store in database as text.
+        """
+        return json.dumps(value)
+
+    def to_python(self, value):
+        """
+        Converts the value into a list.
+        """
+        if not value:
+            value = []
+
+        if isinstance(value, list):
+            if len(value) > LIST_MAX_ITEMS:
+                raise ValidationError(u'list must not contain more than {} items.'.format(LIST_MAX_ITEMS))
+
+            if reduce(lambda a, b: a + len(b), value, 0) > LIST_ITEMS_LENGTH_SUM:
+                raise ValidationError(u'list items cumulative size exceeds {}.'.format(LIST_ITEMS_LENGTH_SUM))
+
+            return value
+
+        try:
+            py_list = json.loads(value)
+            if not isinstance(py_list, list):
+                raise TypeError
+        except (ValueError, TypeError):
+            raise ValidationError(u'Must be a valid list of strings.')
+
+        return py_list
+
+
 class VideoImage(TimeStampedModel):
     """
     Image model for course video.
     """
     course_video = models.OneToOneField(CourseVideo, related_name="video_image")
     image = CustomizableImageField()
+    urls = ListField(max_length=LIST_ITEMS_LENGTH_SUM)
```

Comment:
I think we'll be storing names of the images rather than full CDN urls. What about `generated_images` and image to `active_image` or `selected_image` ? It's your choice.